doc/api/assert.markdown

```diff
@@ -2,8 +2,9 @@
 
     Stability: 2 - Stable
 
-This module is used for writing assertion tests. You can access it with
-`require('assert')`.
+This module is used so that Node.js can test itself. You can access it with
+`require('assert')`. However, it is recommended that you use a userland
+assertion library instead.
```

Comment:
Yeah, at least a docs-only deprecation if that's the way it's supposed to be. @domenic tagged this `tsc-agenda`, so maybe there can be consensus at the next TSC meeting on whether or not `assert` is a general use library or really just intended for use in the Node.js project. If there's general agreement on that, one way or the other, everything flows naturally from it. (Internal use? Doc update only. General purpose? Surely the current surprising behavior should be considered a defect, albeit one that should be fixed carefully and not rushed.)